default/stringtables/st-tool.py

```diff
@@ -0,0 +1,744 @@
+        entries = []
+
+        for item in reference.items():
+            if isinstance(item, StringTableEntry):
+                if item.key in source:
+                    source_item = source[item.key]
+                    entries.append(StringTableEntry(
+                        item.key,
+                        item.keyline,
+                        source_item.value,
+                        item.notes,
+                        source_item.value_times
+                    ))
+                else:
+                    entries.append(StringTableEntry(
+                        item.key,
+                        item.keyline,
+                        None,
+                        item.notes,
+                        item.value_times
+                    ))
+            else:
+                entries.append(item)
+
+        return StringTable(source.fpath, source.language, source.notes, source.includes, entries)
+
+
+class EditServerHandler(BaseHTTPRequestHandler):
+    body = '''
+<html>
+    <head>
+        <title>Editing {source}</title>
+        <style>
+            table {{
+                table-layout: fixed;
+                width: 100%;
+            }}
+            td {{
+                vertical-align: top;
+                white-space: pre-wrap;
+            }}
+            .status-recent td.translation {{
+                background-color: lime;
+                color: black;
+            }}
+            .status-stale td.translation {{
+                background-color: yellow;
+                color: black;
+            }}
+            .status-untranslated td.translation {{
+                background-color: red;
+                color: white;
+            }}
+            .entry td.source {{
+                background-color: lightgrey;
+                color: black;
+            }}
+            form.translator {{
+                width: 100%;
+            }}
+        </style>
+    </head>
+    <body>
+        <h1>Translate stringtable for language &quot;{language}&quot;</h1>
+        <form action="diff" method="post">
+            <fieldset>
+                <legend>Download git diff for submission</legend>
+                <label for="diff-author">Author name</label><input type="input" id="diff-author">
+                <label for="diff-email">Author email</label><input type="input" id="diff-email">
+                <button type="submit" id="save-diff">Download diff</button>
+            </fieldset>
+        </form>
+        <fieldset>
+            <legend>Display settings</legend>
+            <button type="button" id="toggle-recent">Hide recent translations</button>
+            <button type="button" id="toggle-stale">Hide stale translations</button>
+            <button type="button" id="toggle-untranslated">Hide untranslated translations</button>
+        </fieldset>
+        <table>
+            {entries}
+        </table>
+    </body>
+    <script type="text/javascript" src="/jquery.js"></script>
```

Comment:
> Why jquery? Why not pure js? :)

The main reason was that I would like to have some clickable inline editor with buttons so that I don't have to write it from scratch and usuallt I don't do any of this web crap.

I've written this webserver editor thingy in maybe 6 hours with literal no knowledge of javascript while switching from pure, to MooTools to jquery.

Reply:
> Why do you serve it from the server? Can you just give a direct path to file?

Because as far as I know this requires CORS and I'm not familiar enough with that spec and its correct usage.